client/src/app/core/ui-services/banner.service.ts

```
@@ -0,0 +1,58 @@
+import { Injectable } from '@angular/core';
+
+import { BehaviorSubject } from 'rxjs';
+
+export interface BannerDefinition {
+    type?: string;
+    class?: string;
+    icon?: string;
+    text?: string;
+    bgColor?: string;
+    color?: string;
+    link?: string;
+}
+
+/**
+ * A service handling the active banners at the top of the site. Banners are defined via a BannerDefinition
+ * and are removed by reference so the service adding a banner has to store the reference to remove it later
+ */
+@Injectable({
+    providedIn: 'root'
+})
+export class BannerService {
+    public readonly BANNER_HEIGHT = 20;
+
+    public activeBanners: BehaviorSubject<BannerDefinition[]> = new BehaviorSubject<BannerDefinition[]>([]);
+
+    /**
+     * Adds a banner to the list of active banners. Skip the banner if it's already in the list
+     * @param toAdd the banner to add
+     */
+    public addBanner(toAdd: BannerDefinition): void {
+        if (!this.activeBanners.value.find(banner => banner === toAdd)) {
+            const newBanners = this.activeBanners.value.concat([toAdd]);
+            this.activeBanners.next(newBanners);
+        }
+    }
+
+    /**
+     * Replaces a banner with another. Convenience method to prevent flickering
+     * @param toAdd the banner to add
+     * @param toRemove the banner to remove
+     */
+    public replaceBanner(toRemove: BannerDefinition, toAdd: BannerDefinition): void {
+        const newArray = Array.from(this.activeBanners.value);
+        const idx = newArray.findIndex(banner => banner === toRemove);
```

This is too optimistic: Please raise an error, if `toRemove` wan't found.